pkg/terraform/gather/baremetal/ip.go

```diff
@@ -0,0 +1,108 @@
+// Package baremetal contains utilities that help gather Baremetal specific
+// information from terraform state.
+package baremetal
+
+import (
+	"fmt"
+	"github.com/libvirt/libvirt-go"
+	"github.com/openshift-metalkube/kni-installer/pkg/types"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"strings"
+
+	"github.com/openshift-metalkube/kni-installer/pkg/terraform"
+)
+
+// BootstrapIP returns the ip address for bootstrap host. Baremetal relies on a libvirt bootstrap node, that gets it's
+// IP from DHCP.
+func BootstrapIP(tfs *terraform.State, config *types.InstallConfig) (string, error) {
```

Comment:
This is all bit complex, since we have to retrieve the lease information from libvirt.  The libvirt platform generates the IP's: https://github.com/openshift/installer/blob/master/pkg/tfvars/libvirt/libvirt.go

We're also affected by https://github.com/openshift-metal3/dev-scripts/issues/512 here

Reply:
What do we need the BootstrapIP for?  I'm wondering if we can use the static IP configured for the provisioning network in https://github.com/openshift-metal3/kni-installer/pull/100 instead?

Reply:
Debug gathering gets logs from bootstrap if install fails while bootstrap is still up, I think it also uses it as a bastion host to get to the masters in that case as well. 

But we can definitely use the static IP instead of this.

Reply:
I've run into issues getting ironic to work reliably on the boostrap VM, but I may try breaking #100 into two or three PRs, then we can enable the provisioning nic so that this PR can use it, then potentially host the images to allow fixing https://github.com/openshift-metal3/kni-installer/issues/58, and finally switch the ironic endpoint over to the bootstrap VM.

@stbenjam let me know if the above decoupling of PRs would be helpful, if so I'll rebase and open some additional PRs.

Reply:
I think that makes sense. I don't think we need install gather to open the first PR to openshift/installer but we do need it eventually.  

Could we move the masters to static IP's as well? Or do you think the approach of getting them out of ironic introspection data is reliable?

Reply:
Yes I think we'll need to switch the masters to static IPs to enable self-hosted ironic where we'll need to run dnsmasq and the other ironic bits on a master to deploy the workers - ref discussion on https://github.com/metal3-io/metal3-docs/pull/39

I'm not yet sure what the best way to inject that configuration is though, e.g we could use ignition, something run via the BMO or perhaps the MCO, and there's the nmstate operator to investigate too.